src/dsi.rs

```diff
@@ -0,0 +1,672 @@
+//! Display Serial Interface
+//!
+//! Interface with MIPI D-PHY
+
+use crate::ltdc::DisplayConfig;
+use crate::rcc::{Clocks, Enable};
+use crate::{pac::DSI, time::Hertz};
+use core::cmp::{max, min};
+use embedded_display_controller::dsi::{DsiHostCtrlIo, DsiReadCommand, DsiWriteCommand};
+
+const DSI_TIMEOUT_MS: usize = 100;
+
+pub struct DsiHost {
+    dsi: DSI,
+    channel: DsiChannel,
+    cycles_1ms: u32,
+}
+
+#[derive(Copy, Clone)]
+pub enum DsiChannel {
+    Ch0 = 0b00,
+    Ch1 = 0b01,
+    Ch2 = 0b10,
+    Ch3 = 0b11,
+}
+
+#[derive(Debug)]
+pub enum Error {
```

Comment:
All enums should implement these traits:
```suggestion
#[cfg_attr(feature = "defmt", derive(defmt::Format))]
#[derive(Clone, Copy, Debug, PartialEq, Eq)]
pub enum Error {
```